### PR TITLE
Added feature to create new groups within a course. Added test.

### PR DESCRIPTION
**Pull request by cookieKeks**
You can now add new groups to a gourse. If the group already exists, the user will be asked if he still wants to create the group. To be able to ask the user, I used a method from IOUtils which is originally from the ilias-client (The original repository). Also added a test.

### CODE REVIEW

File: src/main/java/com/github/mavogel/ilias/lib/wrapper/soap/SoapEndpoint.java [hunk 1]
```diff
@@ -365,4 +367,59 @@ public void setRegistrationDatesOnGroups(final List<IliasNode> groups, final Loc
...
+            LOG.debug("With ref_id: " + result);
+            return true;
+        } catch (RemoteException e) {
+            LOG.error("Could not add group " + groupName);
+            return false;
+        }
+    }
+
+    //TODO: Taken from IOUtils from the ilias-client. Maybe move the class to this lib and put this method there?
+    private static boolean readAndParseUserConfirmation() {
```
Comment (mavogel) [inline src/main/java/com/github/mavogel/ilias/lib/wrapper/soap/SoapEndpoint.java]: Remove this method and let it in the `ilias-client`

File: src/main/java/com/github/mavogel/ilias/lib/wrapper/soap/SoapXMLUtils.java [hunk 2]
```diff
@@ -329,4 +329,18 @@ public static int parseGroupMemberRoleId(final String localRolesForGroupXML) thr
...
                 u.getChild("Email").getText().trim())).collect(Collectors.toList());
     }
+
+    /**
+     * Generates an XML string needed to create a new group via  the SOAP interface.
+     * @param groupName The name of the group
+     * @return The XML string
+     */
+    public static String generateAddGroupXMLFromName(String groupName) {
+        return "<?xml version=\"1.0\" encoding=\"utf-8\"?>" +
```
Comment (mavogel) [inline src/main/java/com/github/mavogel/ilias/lib/wrapper/soap/SoapXMLUtils.java:339]: It makes more sense to use a `StringBuilder` here

File: src/main/java/com/github/mavogel/ilias/lib/wrapper/soap/SoapEndpoint.java [hunk 3]
```diff
@@ -365,4 +367,59 @@ public void setRegistrationDatesOnGroups(final List<IliasNode> groups, final Loc
...
+     * courses of the user and not just in the given one. We cannot restrict this search, thus
+     * the user has to decide. (@see {@link com.github.mavogel.ilias.lib.wrapper.IliasEndpoint})
+     *
+     * @param course    The course the group shall be created in
+     * @param groupName The name of the group
+     * @return True iff the group has been successfully been created
+     * @throws Exception in case of a failure. Detailed logs are written.
+     */
+    @Override
+    public boolean addGroup(IliasNode course, String groupName) throws Exception {
```
Comment (mavogel) [inline src/main/java/com/github/mavogel/ilias/lib/wrapper/soap/SoapEndpoint.java]: I'd suggest 2 Methods though if it is not possible to determine in which course the group already exists

1. Check if a group already exists
```java
public boolean groupExists(String groupName)
```
2. Adds a group within the given course. Throws an exception if it fails
```java
public void addGroup(IliasNode course, String groupName) throws Exception
```

Then handle the User IO in the calling client, like e.g. the `IliasClient` or your library

**Review by mavogel — CHANGES_REQUESTED** [review]

**Comment by mavogel** [comment]
BTW I'll fix the `travis` build then. Its failure has nothing to do with your enhancement.

File: src/main/java/com/github/mavogel/ilias/lib/wrapper/soap/SoapEndpoint.java (hunk 1 above)
Comment (cookieKeks) [reply src/main/java/com/github/mavogel/ilias/lib/wrapper/soap/SoapEndpoint.java]: Done

File: src/main/java/com/github/mavogel/ilias/lib/wrapper/soap/SoapXMLUtils.java (hunk 2 above)
Comment (cookieKeks) [reply src/main/java/com/github/mavogel/ilias/lib/wrapper/soap/SoapXMLUtils.java:339]: I have used Stringbuilder initially, but then Intellij suggests to convert it to a String. The situation is also described here: 
https://stackoverflow.com/questions/42855964/how-to-prevent-notice-stringbuilder-sb-can-be-replaced-with-string-in-int 
and here 
https://intellij-support.jetbrains.com/hc/en-us/community/posts/115000093250-Convert-StringBuilder-to-String-concatenation-suggestion.
Thus I followed the suggestion. However, if you still want to, I can of course use a StringBuilder.

File: src/main/java/com/github/mavogel/ilias/lib/wrapper/soap/SoapEndpoint.java (hunk 3 above)
Comment (cookieKeks) [reply src/main/java/com/github/mavogel/ilias/lib/wrapper/soap/SoapEndpoint.java]: Yes that makes more sense. Done too.

**Comment by cookieKeks** [comment]
Applied the suggested changes about extracting the group-exist-check and the removal of the IOUtils method.

File: src/main/java/com/github/mavogel/ilias/lib/wrapper/soap/SoapXMLUtils.java (hunk 2 above)
Comment (mavogel) [reply src/main/java/com/github/mavogel/ilias/lib/wrapper/soap/SoapXMLUtils.java:339]: Just for readability, but let's leave it with concatenation. It's fine

**Review by mavogel — APPROVED** [review]
LGTM thx 😃 
I fixed `travis` locally and need to investigate more why it didn't work before. Maybe change the setup.

**Pull request merged**